src/mlpack/tests/gmm_test.cpp

```diff
@@ -11,7 +11,6 @@
  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
  */
 #include <mlpack/core.hpp>
-#include <mlpack/core/math/ccov.hpp>
```

Comment:
Adding this line throws unexpected error , snap shot of which i have added. If i remove this Line the build is successful.

Reply:
Also after long hours of research, I found that `make -j1` throws error, `make -j4` works perfctly alright that means some issue with the order of compilation, but still i am unable to reach a concrete reason.

Reply:
`ColumnCovariance()`, need `Log::Fatal` and hence include, `mlpack/prereqs.hpp` and thus we cannot inlcude `ccov.hpp` in `mlpack/prereqs.hpp` because it throws an error 

`make -j4` runs parallely and would manage to compile the files in correct order, but while going with `make -j1`, the build fails since it disturbs the order of compilation.

Therfore i have added it in `mlpack/core.hpp`, Along with other `mlpack/data/math/*` , I guess it won't make much difference since core is included at all places and thus, it would be ok to include the file through `core.hpp`

Reply:
Ah, you are right, sorry---`core.hpp` would be where we have to put it.  Nice debugging :+1: